lib/detect-testing-library-utils.ts

```diff
@@ -13,48 +18,84 @@ export function detectTestingLibraryUtils<
   TRuleListener extends TSESLint.RuleListener = TSESLint.RuleListener
 >(
   ruleCreate: (
-    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    context: Readonly<
+      TSESLint.RuleContext<TMessageIds, TOptions> & {
+        settings: TestingLibrarySettings;
+      }
+    >,
     optionsWithDefault: Readonly<TOptions>,
     detectionHelpers: Readonly<DetectionHelpers>
   ) => TRuleListener
 ) {
   return (
-    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    context: Readonly<
+      TSESLint.RuleContext<TMessageIds, TOptions> & {
+        settings: TestingLibrarySettings;
+      }
+    >,
     optionsWithDefault: Readonly<TOptions>
-  ): TRuleListener => {
-    let isImportingTestingLibrary = false;
+  ): TSESLint.RuleListener => {
```

Comment:
I've replaced returned type from `TRuleListener` to `TSESLint.RuleListener` since `detectTestingLibraryUtils` modify the received rules listener to add or remove some so it can end being a different set of rule listener.